libraries/botbuilder-storage-mssql/src/mssqlStorage.ts

```diff
@@ -0,0 +1,105 @@
+import { Storage, StoreItem, StoreItems } from 'botbuilder-core';
+import { config as msconfig, ConnectionPool, NVarChar, IResult, Int } from 'mssql';
+import { MSSQLOptions } from './schema';
+
+/**
+ * @module @botbuildercommunity/storage-mssql
+ */
+
+function getConnectionPool(connection): ConnectionPool {
+    try {
+        return new ConnectionPool(connection);
+    }
+    catch(e) {
+        console.error(e);
+    }
+}
+
+export class MSSQLStorage implements Storage {
+    private options: MSSQLOptions;
+    private connection: msconfig;
+    public constructor(options: MSSQLOptions) {
+        this.options = options;
+        this.connection = {
+            user: options.dbuser,
+            password: options.dbpassword,
+            server: options.dbserver,
+            database: options.db
+        };
+    }
+    public async read(keys: string[]): Promise<StoreItems> { //Should this be an array of numbers?
+        if(keys == null || keys.length === 0) {
```

Comment:
keys === null? Should this be strict checking, not sure what the values of keys can be.

Reply:
I lazy check `null`. My assumption is `keys` can't actually be `undefined` but I haven't tested that theory yet.

Reply:
Did a little research, and I think I'm going to leave this with the lazy check for now.